pkg/handlers/webhookreceiver.go

```diff
@@ -268,34 +268,36 @@ func (h *WebhookReceiverHandler) updateNotificationStatus(n *oav1alpha1.Notifica
 		} else {
 			// Status exists, update it
 			// When the alert is already firing
-			firingCondition := status.Conditions.GetCondition(oav1alpha1.ConditionAlertFiring).Status
-			if firingCondition == corev1.ConditionTrue {
-				firedConditionTime := status.Conditions.GetCondition(oav1alpha1.ConditionAlertFiring).LastTransitionTime
-				resolvedConditionTime := status.Conditions.GetCondition(oav1alpha1.ConditionAlertResolved).LastTransitionTime
-				if firing {
-					// Status transition is Firing to Firing
-					// Do not update the condition for AlertFiring and AlertResolved
-					// Only update the timestamp for the ServiceLogSent
-					status.SetStatus(oav1alpha1.ConditionAlertFiring, "Alert is firing", corev1.ConditionTrue, firedConditionTime)
-					status.SetStatus(oav1alpha1.ConditionAlertResolved, "Alert resolved", corev1.ConditionFalse, resolvedConditionTime)
-					status.SetStatus(oav1alpha1.ConditionServiceLogSent, "Service log sent", corev1.ConditionTrue, timeNow)
+			firingCondition := status.Conditions.GetCondition(oav1alpha1.ConditionAlertFiring)
+			if firingCondition != nil {
```

Comment:
This refactoring has no `else`. There should never be a state where we don't update status at all.

Reply:
Thanks I've added the else block as suggested